src/codec.rs

```diff
@@ -326,7 +326,7 @@ mod tests {
 
     #[test]
     fn decodes_zero_length_message() {
-        let content_type = "Content-Type: application/vscode-jsonrpc; charset=utf-8";
+        let content_type = "application/vscode-jsonrpc; charset=utf-8";
         let encoded = encode_message(Some(content_type), "");
```

Comment:
I noticed a small bug here (that doesn't trigger a change in the test, but still).

The `encode_message` function already applies the `"Content-Type: "` prefix to the argument.

Reply:
Nice catch! Thanks for fixing this.